src/tools/evaluate.ts

```
@@ -42,22 +42,38 @@ const evaluate = defineTabTool({
-      const func = new Function() as any;
-      func.toString = () => params.function;
-      const result = locator ? await locator.evaluate(func) : await tab.page.evaluate(func);
-      response.addResult(JSON.stringify(result, null, 2) || 'undefined');
+      // Evaluate `(${source})` inside the page so both a plain expression
+      // (e.g. `document.title`) and a function form (e.g. `() => ...` or
+      // `(element) => ...`) are accepted. Running the source in-page also avoids
+      // the private `_evaluateFunction` helper removed in playwright-core 1.59.
+      let evalResult: { result: unknown, isFunction: boolean };
+      if (locator) {
+        evalResult = await locator.evaluate(async (element, source) => {
+          const value = eval(`(${source})`);
```

Biome (2.5.1)

[error] eval() exposes to security risks and performance issues.

(lint/security/noGlobalEval)